File: festival-web/src/main/java/cn/bdqfork/web/annotation/GetMapping.java
package cn.bdqfork.web.annotation;

import io.vertx.core.http.HttpMethod;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * 用于GET请求
 *
 * @author bdq
 * @since 2020/1/21
 */
@RouteMapping(method = HttpMethod.GET)
@Documented
@Retention(RUNTIME)
@Target({ElementType.ANNOTATION_TYPE, ElementType.METHOD})
public @interface GetMapping {
    String value();
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/annotation/GetMapping.kt
package cn.bdqfork.kotlin.web.route.annotation

import io.vertx.core.http.HttpMethod

/**
 * 用于GET请求
 *
 * @author bdq
 * @since 2020/1/21
 */
@RouteMapping(method = HttpMethod.GET)
@MustBeDocumented
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.FUNCTION)
annotation class GetMapping(val value: String, val timeout: Long = -1)